main/lsp/requests/hover.cc

```diff
@@ -24,11 +24,38 @@ string methodSignatureString(const core::GlobalState &gs, const core::TypePtr &r
     return contents;
 }
 
-unique_ptr<MarkupContent> formatRubyCode(MarkupKind markupKind, string str) {
-    if (markupKind == MarkupKind::Markdown && str.length() > 0) {
-        str = fmt::format("```ruby\n{}\n```", str);
+unique_ptr<MarkupContent> formatRubyCode(MarkupKind markupKind, string_view sigString,
+                                         optional<string_view> optionalDocString) {
+    string_view docString = optionalDocString.value_or("");
+    // Get rid of at most 1 trailing newline
+    if (docString.length() > 0 && docString.back() == '\n') {
+        docString.remove_suffix(1);
     }
-    return make_unique<MarkupContent>(markupKind, move(str));
+    if (sigString.length() > 0 && sigString.back() == '\n') {
+        sigString.remove_suffix(1);
+    }
+
+    string content = "";
+
+    // Add docs
+    absl::StrAppend(&content, docString);
+    if (docString.length() > 0)
+        absl::StrAppend(&content, "\n");
+
+    // Add sig
+    string formattedSigString = "";
+    if (markupKind == MarkupKind::Markdown && sigString.length() > 0) {
+        formattedSigString = fmt::format("```ruby\n{}\n```", sigString);
+    } else {
+        absl::StrAppend(&formattedSigString, sigString);
+    }
+    absl::StrAppend(&content, formattedSigString);
+
+    // Add newline at the end if needed
```

Comment:
Why is a newline needed?

Reply:
This was to handle docs without a newline + empty sig string. Not required anymore since we handle trimming in different functions